engine/linksystem.go

```diff
@@ -95,7 +96,7 @@ func (e *Engine) mkLinkSystem() ipld.LinkSystem {
 			// deletes all indexes for the contextID in the removal
 			// advertisement.  Only if the removal had no contextID would the
 			// indexer ask for entry chunks to remove.
-			mhIter, err := e.mhLister(ctx, key)
+			mhIter, err := e.mhLister(ctx, key.Provider, key.ContextID)
```

Comment:
yes, good spot. I tweaked the behaviour a bit so that provider is resolved to default in the very beginning and hence no other functions should assume that the provider can be an empty string